packages/fields/src/fields/featured-image/index.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import type { Field } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import type { BasePost } from '../../types';
+import { __ } from '@wordpress/i18n';
+import { FeaturedImageEdit } from './featured-image-edit';
+import { FeaturedImageView } from './featured-image-view';
+
+const featuredImageField: Field< BasePost > = {
+	id: 'featured_media',
```

Comment:
What's the `type` for this field? We should make progress on making the `type` obligatory for fields and every new field we add should introduce the corresponding type if there's one. Is it `attachment`, `image`, etc.?

Reply:
An additional thought: I think we'd want to have two behaviors for images:

- an attachment, like in this PR
- an embedded image, like the feature image is in the document inspector

How that would look like?

Reply:
To clarify, are you suggesting adding the type of entity that the field will display or edit?

If so, I think that we need to discuss this further. I believe the type field should be specific to only one field component.

In the future, the type field will be essential for registering the dedicated field in the Redux store, establishing a unique 1:1 relationship between the type and the field.

To elaborate further, we are planning to develop the `registerFieldType` API, which will be essential for registering new custom fields. The type value will distinguish one custom field from another.

With c448c494b25d9bf31b5e3c0b76c07e2f9a88bd04, I added `featured-image` as type.

In any case, I'm happy to discuss this further. It is really important that we are aligned on this topic! 🙇

Reply:
We are discussing the same topic [here](https://github.com/WordPress/gutenberg/pull/65196#discussion_r1782971224)

Reply:
I think the "type" here should be "media" and ideally we wouldn't need to define an "edit" function in that case. All media in WordPress could use that type. That said, that's dependent on the possibility to register field types which we don't have at the moment. So I'd suggest for this PR staying with something low level like "number" or "string".

Reply:
Fixed with 9cc6d80958c9a240524cbe0e715b81a38030e37a

Reply:
I actually suggested "integer" or "text" instead of "media" for now. I don't think we should be using a type that doesn't have any registered definition.